jkq/ddsim/circuitsimulator.py

```diff
@@ -0,0 +1,113 @@
+"""Backend for DDSIM."""
+
+import logging
+import time
+import uuid
+
+from qiskit.providers import BackendV1, Options
+from qiskit import qobj, QiskitError
+from qiskit.providers.models import BackendConfiguration, BackendStatus
+from qiskit.result import Result
+
+from .jkqjob import JKQJob
+from .jkqerror import JKQSimulatorError
+
+from jkq.ddsim import pyddsim
+
+logger = logging.getLogger(__name__)
+
+
+class CircuitSimulator(BackendV1):
+    """Python interface to JKQ DDSIM"""
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        return Options(
+            shots=None,
+            return_statevector=True
+        )
+
+    def __init__(self, configuration=None, provider=None):
+        conf = {
+            'backend_name': 'qasm_simulator',
+            'backend_version': '1.4.3',
+            'url': 'https://github.com/iic-jku/ddsim',
+            'simulator': True,
+            'local': True,
+            'description': 'JKQ DDSIM C++ simulator',
+            'basis_gates': ['id', 'u0', 'u1', 'u2', 'u3', 'cx', 'x', 'y', 'z', 'h', 's', 't', 'snapshot'],
```

Comment:
We do support a much broader list of basis gates. See https://github.com/iic-jku/qfr/blob/master/jkq/qfr/QiskitImport.hpp for all the gates natively supported.
It should be possible to also allow multi-controlled gates here.